### PR TITLE
fix: preserve infinite losses in weighted reductions

### LINKED ISSUE #1601: [Losses] _divide_no_nan turns an infinite loss into 0.0, so broken trials win Auto search
### What happened + What you expected to happen

What happened:
A diverged trial reported loss = 0.0. Its predictions were ±inf, so the loss
should have been inf. Because 0.0 is the lowest possible value, the Auto
search treated the broken trial as the best one, kept it through every ASHA
rung, and returned it as the winner. No warning, no exception. best_config
stored the diverged configuration.

What I expected to happen:
A diverged trial should report inf, or at least a very large loss, so it
ranks worst in the search. The winner should be one of the trials that
actually converged.

### Versions / Dependencies

```
neuralforecast 3.2.1
torch 2.13.0+cu130
ray 2.57.0
python 3.12.3
Linux
```

### Reproduction script

```python
import torch
from neuralforecast.losses.pytorch import MAE

y = torch.tensor([[[10.0], [20.0]]])
print(MAE()(y=y, y_hat=torch.full_like(y, float("inf"))))

# actual:   tensor(0.)
# expected: tensor(inf)
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by kiwoongyoon**
Title: fix: preserve infinite losses in weighted reductions

## Summary

- Preserve signed infinities in `_divide_no_nan` instead of replacing them with zero.
- Ignore zero-weight entries before multiplying weighted losses, preventing masked infinities from producing `NaN`.
- Cover unmasked and masked positive/negative infinite predictions with an MAE regression test.

Fixes #1601

## Test evidence

- `pytest tests/test_losses/test_pytorch.py::test_MAE_infinite_predictions -q` — 1 passed
- `pytest tests/test_losses/test_pytorch.py -q` — 16 passed
- `ruff check neuralforecast/losses/pytorch.py tests/test_losses/test_pytorch.py` — passed
- `git diff --check` — passed

## Risks or notes for maintainers

`_weighted_mean` is shared by point losses, so zero-weight non-finite values are now ignored consistently wherever masks or zero horizon weights are used. No public API or dependency changes are included.

### CODE REVIEW

**Comment by marcopeix** [comment]
@kiwoongyoon Thanks for raising the issue and drafting a PR. I think I have a more complete fix in #1603

**Comment by marcopeix** [comment]
Closing since merging #1603
